core/src/main/java/com/netflix/iceberg/hadoop/HadoopInputFile.java

```diff
@@ -76,6 +76,10 @@ public static HadoopInputFile fromStatus(FileStatus stat, Configuration conf) {
     }
   }
 
+  static HadoopInputFile fromFsPath(FileSystem fs, Path path, Configuration conf) {
```

Comment:
Instead of adding this, I think `HadoopInputFile` should make its constructor protected to allow you to create a subclass.

The reason why we use `Path` and `Configuration` is to ensure that the file system always matches the path's scheme. Adding a new factory method that allows users to supply a file system creates a path around that guarantee. That's legitimate for your use case, but you'll need to implement `HadoopTableOperations` anyway, so I think a safer way to solve the problem for you is to subclass `HadoopInputFile` and expose this method in your private subclass.

Reply:
I should note that for the limited use in this PR I think it is okay because this is package-private. It would just be better to solve your overall use case with a custom `TableOperations` implementation.